extensions-contrib/kafka-emitter/src/main/java/io/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -0,0 +1,193 @@
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+public class KafkaEmitter implements Emitter {
+  private static Logger log = new Logger(KafkaEmitter.class);
+
+  private final static String DEFAULT_KEY_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+  private final static String DEFAULT_VALUE_SERIALIZER = "org.apache.kafka.common.serialization.StringSerializer";
+  private final static int DEFAULT_RETRIES = 3;
+  private long queueMemoryBound = 33554432L; // same with default value of kafka producer's buffer.memory
+  private final AtomicLong metricLost;
+  private final AtomicLong alertLost;
+  private final AtomicLong invalidLost;
+
+  private final KafkaEmitterConfig config;
+  private final Producer<String, String> producer;
+  private final Callback producerCallback;
+  private final ObjectMapper jsonMapper;
+  private final MemoryBoundLinkedBlockingQueue<String> metricQueue;
+  private final MemoryBoundLinkedBlockingQueue<String> alertQueue;
+  private final ScheduledExecutorService scheduler;
+
+  public KafkaEmitter(KafkaEmitterConfig config, ObjectMapper jsonMapper) {
+    this.config = config;
+    this.jsonMapper = jsonMapper;
+    this.producer = setKafkaProducer();
+    this.producerCallback = setProducerCallback();
+    this.metricQueue = new MemoryBoundLinkedBlockingQueue<>(queueMemoryBound);
+    this.alertQueue = new MemoryBoundLinkedBlockingQueue<>(queueMemoryBound);
+    this.scheduler = Executors.newScheduledThreadPool(3);
+    this.metricLost = new AtomicLong(0L);
+    this.alertLost = new AtomicLong(0L);
+    this.invalidLost = new AtomicLong(0L);
+  }
+
+  private Callback setProducerCallback() {
+    return new Callback() {
+      @Override
+      public void onCompletion(RecordMetadata recordMetadata, Exception e) {
+        if(e != null) {
+          log.debug("Event send failed [%s]", e.getMessage());
+          if(recordMetadata.topic().equals(config.getMetricTopic())) {
+            metricLost.incrementAndGet();
+          } else if (recordMetadata.topic().equals(config.getAlertTopic())) {
+            alertLost.incrementAndGet();
+          } else {
+            invalidLost.incrementAndGet();
+          }
+        }
+      }
+    };
+  }
+
+  private Producer<String, String> setKafkaProducer() {
+    Properties props = new Properties();
+    props.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, config.getBootstrapServers());
+    props.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, DEFAULT_KEY_SERIALIZER);
+    props.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, DEFAULT_VALUE_SERIALIZER);
+    props.put(ProducerConfig.RETRIES_CONFIG, DEFAULT_RETRIES);
+    props.putAll(config.getKafkaProducerConfig());
+    queueMemoryBound = (props.containsKey(ProducerConfig.BUFFER_MEMORY_CONFIG) ?
+        Long.parseLong(props.getProperty(ProducerConfig.BUFFER_MEMORY_CONFIG)) : queueMemoryBound);
+
```

Comment:
@dkhwangbo I mean set the `Thread.currentThread().setContextClassLoader(getClass().getClassLoader())` before init a kafkaProducer in KafkaEmitter.setKafkaProducer() will works. This will make the KafkaProducer init with right classloader(URLClassLoader).

By the way, may I ask is your extensions folder in your classpath? If your extensions folder is in your classpath, then maybe there won't be this problem.